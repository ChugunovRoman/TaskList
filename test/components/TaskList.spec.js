'use strict';

import React from 'react';
import sinon from 'sinon';
import TaskList from '../../src/components/TaskList';

describe('Components: <TaskList />', () => {

    it('should renders without crashing', () => {
        const onComplete = sinon.spy();

        const props = {
            tasks: [
                {
                    id: '0',
                    name: 'Task 1',
                    description: 'Description 1'
                },
                {
                    id: '1',
                    name: 'Task 2',
                    description: 'Description 2',
                    completed: true
                },
                {
                    id: '2',
                    name: 'Task 3',
                    description: 'Description 3',
                    completed: true
                },
                {
                    id: '3',
                    name: 'Task 4',
                    description: 'Description 4'
                }
            ],
            onComplete: onComplete
        };

        const wrapper = mount (
            <TaskList {...props} />
        );

        expect(wrapper.find('Task')).toHaveLength(4);

        expect(wrapper).toMatchSnapshot();
    });

});
