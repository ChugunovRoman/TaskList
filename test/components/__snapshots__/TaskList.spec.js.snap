// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: <TaskList /> should renders without crashing 1`] = `
<TaskList
  onComplete={[Function]}
  tasks={
    Array [
      Object {
        "description": "Description 1",
        "id": "0",
        "name": "Task 1",
      },
      Object {
        "completed": true,
        "description": "Description 2",
        "id": "1",
        "name": "Task 2",
      },
      Object {
        "completed": true,
        "description": "Description 3",
        "id": "2",
        "name": "Task 3",
      },
      Object {
        "description": "Description 4",
        "id": "3",
        "name": "Task 4",
      },
    ]
  }
>
  <div
    className="app__tasks tasks"
  >
    <Task
      description="Description 1"
      id="0"
      name="Task 1"
      onComplete={[Function]}
    >
      <div
        className="task"
      >
        <div
          className="task__name"
          onClick={[Function]}
        >
          <p>
            Task 1
          </p>
        </div>
        <div
          className="task__desc desc"
        >
          <p
            className="desc__text"
          >
            Description 1
          </p>
          <button
            className="desc__btn btn"
            disabled=""
            onClick={[Function]}
            value="0"
          >
            Завершить
          </button>
        </div>
      </div>
    </Task>
    <Task
      completed={true}
      description="Description 2"
      id="1"
      name="Task 2"
      onComplete={[Function]}
    >
      <div
        className="task task_completed"
      >
        <div
          className="task__name"
          onClick={[Function]}
        >
          <p>
            Task 2
          </p>
        </div>
        <div
          className="task__desc desc"
        >
          <p
            className="desc__text"
          >
            Description 2
          </p>
          <button
            className="desc__btn btn"
            disabled="disabled"
            onClick={[Function]}
            value="1"
          >
            Завершить
          </button>
        </div>
      </div>
    </Task>
    <Task
      completed={true}
      description="Description 3"
      id="2"
      name="Task 3"
      onComplete={[Function]}
    >
      <div
        className="task task_completed"
      >
        <div
          className="task__name"
          onClick={[Function]}
        >
          <p>
            Task 3
          </p>
        </div>
        <div
          className="task__desc desc"
        >
          <p
            className="desc__text"
          >
            Description 3
          </p>
          <button
            className="desc__btn btn"
            disabled="disabled"
            onClick={[Function]}
            value="2"
          >
            Завершить
          </button>
        </div>
      </div>
    </Task>
    <Task
      description="Description 4"
      id="3"
      name="Task 4"
      onComplete={[Function]}
    >
      <div
        className="task"
      >
        <div
          className="task__name"
          onClick={[Function]}
        >
          <p>
            Task 4
          </p>
        </div>
        <div
          className="task__desc desc"
        >
          <p
            className="desc__text"
          >
            Description 4
          </p>
          <button
            className="desc__btn btn"
            disabled=""
            onClick={[Function]}
            value="3"
          >
            Завершить
          </button>
        </div>
      </div>
    </Task>
  </div>
</TaskList>
`;
